tests/workflow_task_reported_problems_test.go

```diff
@@ -0,0 +1,360 @@
+		RetryPolicy:            s.activityRetryPolicy,
+	}), s.simpleActivity).Get(ctx, &ret)
+	if err != nil {
+		return "", err
+	}
+
+	if s.shouldFail.Load() {
+		panic("forced-panic-to-fail-wft")
+	}
+
+	return "done!", nil
+}
+
+func (s *WFTFailureReportedProblemsTestSuite) TestWFTFailureReportedProblems_SetAndClear() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	s.OverrideDynamicConfig(dynamicconfig.NumConsecutiveWorkflowTaskProblemsToTriggerSearchAttribute, 2)
+	s.shouldFail.Store(true)
+
+	s.Worker().RegisterWorkflow(s.simpleWorkflow)
+
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        testcore.RandomizeStr("wf_id-" + s.T().Name()),
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, s.simpleWorkflow)
+	s.NoError(err)
+
+	// Make sure the workflow has started and had an activity task scheduled and finished
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.Equal(t, enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING, description.WorkflowExecutionInfo.Status)
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Check if the search attributes are not empty and has TemporalReportedProblems
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.NotNil(t, description.WorkflowExecutionInfo.SearchAttributes)
+		require.NotEmpty(t, description.WorkflowExecutionInfo.SearchAttributes.IndexedFields)
+		require.NotNil(t, description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems])
+
+		// Decode the search attribute in keyword list format
+		searchValBytes := description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems]
+		searchVal, err := searchattribute.DecodeValue(searchValBytes, enumspb.INDEXED_VALUE_TYPE_KEYWORD_LIST, false)
+		require.NoError(t, err)
+		require.NotEmpty(t, searchVal)
+		require.Equal(t, "category=WorkflowTaskFailed", searchVal.([]string)[0])
+		require.Equal(t, "cause=WorkflowWorkerUnhandledFailure", searchVal.([]string)[1])
+
+		// Validate attempt number after verifying search attribute values
+		require.GreaterOrEqual(t, description.GetPendingWorkflowTask().Attempt, int32(2))
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Check if the search attributes are queryable
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		queriedWorkflows, err := s.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			Query:     "TemporalReportedProblems IS NOT NULL",
+			PageSize:  100,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(queriedWorkflows.Executions))
```

Comment:
Can just delete this, you're not testing the visibility system, the fact that the second query returns a list of size 1 is enough to give you the coverage you need.